backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarWriteFilesExec.scala

```diff
@@ -0,0 +1,151 @@
+import org.apache.spark.sql.catalyst.catalog.CatalogTypes.TablePartitionSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.connector.write.WriterCommitMessage
+import org.apache.spark.sql.execution.datasources.{BasicWriteTaskStats, ExecutedWriteSummary, FileFormat, PartitioningUtils, WriteFilesExec, WriteFilesSpec, WriteTaskResult}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+import shaded.parquet.com.fasterxml.jackson.databind.ObjectMapper
+
+import scala.collection.mutable
+
+class VeloxColumnarWriteFilesExec(
+    child: SparkPlan,
+    fileFormat: FileFormat,
+    partitionColumns: Seq[Attribute],
+    bucketSpec: Option[BucketSpec],
+    options: Map[String, String],
+    staticPartitions: TablePartitionSpec)
+  extends WriteFilesExec(
+    child,
+    fileFormat,
+    partitionColumns,
+    bucketSpec,
+    options,
+    staticPartitions) {
+
+  override def supportsColumnar(): Boolean = true
+
+  override def doExecuteWrite(writeFilesSpec: WriteFilesSpec): RDD[WriterCommitMessage] = {
+    assert(child.supportsColumnar)
+
+    // We need to pass the WritePath to the Velox TableWriter in the doTransform
+    // method of the WriteTransformer. However, the WritePath is not accessible
+    // during the planning phase in the WriteTransformer, and can only be obtained
+    // during the actual execution, specifically in the doExecuteWrite method of
+    // ColumnarWriteFilesExec, where it is available within the WriteFilesSpec.
+    // Therefore, we use this hack method to pass the writePath.
+    child.session.sparkContext.setLocalProperty("writePath", writeFilesSpec.description.path)
+    child.executeColumnar().mapPartitionsInternal {
+      iter =>
+        // Currently, the cb contains three columns: row, fragments, and context.
+        // The first row in the row column contains the number of written numRows.
+        // The fragments column contains detailed information about the file writes.
+        // The json can be as following:
+        // {
+        //     "inMemoryDataSizeInBytes":0,
+        //     "containsNumberedFileNames":true,
+        //     "onDiskDataSizeInBytes":307,
+        //     "fileWriteInfos":[
+        //         {
+        //             "fileSize":307,
+        //             "writeFileName":
+        //                "Gluten_Stage_1_TID_2_0_2_d1db3b31-4f99-41cb-a4e7-3b8607506168.parquet",
+        //             "targetFileName":
+        //                "Gluten_Stage_1_TID_2_0_2_d1db3b31-4f99-41cb-a4e7-3b8607506168.parquet"
+        //         }
+        //     ],
+        //     "writePath":"file:/home/gluten/spark-warehouse/inserttable/part1=1/part2=1",
+        //     "rowCount":1,
+        //     "targetPath":"file:/home/gluten/spark-warehouse/inserttable/part1=1/part2=1",
+        //     "updateMode":"NEW",
+        //     "name":"part1=1/part2=1"
+        // }
+        if (iter.hasNext) {
```

Comment:
Instead of adding if-else, can we add `assert(iter.hasNext)` and then `val cb = iter.next()` ?